src/executor/Scheduler.ts

```diff
@@ -127,18 +117,28 @@ export class Scheduler {
         }
         const nextOpenAPIYamlPlugin = openAPIYamlSchedulingResult.scheduledPlugin;
 
-        const vendorSchedulingResult = this.getAndScheduleNextJob(this.vendorJobs, 'vendor', 'vendor');
-        if (vendorSchedulingResult.backoff) {
-            return;
-        }
-        const nextVendorPlugin = vendorSchedulingResult.scheduledPlugin;
-
         const compressCssSchedulingResult = this.getAndScheduleNextJob(this.compressCssJobs, 'compressCss', 'css');
         if (compressCssSchedulingResult.backoff) {
             return;
         }
         const nextCompressCssPlugin = compressCssSchedulingResult.scheduledPlugin;
 
+        let nextTsE2EPlugin: string | null | undefined = null;
+        if (!this.isProduction) {
+            const tsE2ESchedulingResult = this.getAndScheduleNextJob(this.tsE2EJobs, 'tsE2E', 'tsE2E');
+            if (tsE2ESchedulingResult.backoff) {
+                return;
+            }
+            nextTsE2EPlugin = tsE2ESchedulingResult.scheduledPlugin;
+        }
+
+        const tsSchedulingResult = this.getAndScheduleNextJob(this.tsJobs, 'ts', 'ts');
+        if (tsSchedulingResult.backoff) {
```

Comment:
Does the problem occure when cplace-asc is started in prod mode that it always has tsSchedulingResult.backoff=true ?

could it additionally help changing to ?
  `      if (!this.isProduction && tsSchedulingResult.backoff) {`

Reply:
The problem occurs if VendorCompiler (plugin NPM modules aren't installed) is not done yet, and TSCompiler is running. Because some of the NPM modules aren't installed yet, TSCompiler is throwing errors. I reordered now the compileRequests, to run the vendor first and TS last. Seems to be working fine. Check [build](https://app.circleci.com/pipelines/github/collaborationFactory/cplace-freshfields/7497/workflows/741312bc-9749-4d7f-99c6-010b21c9df1c/jobs/7984/steps)

Reply:
ah ok so vendor compilation step just has to be done upfront. Should we add a comment for that?

Reply:
Good point. I will add a comment